mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Image.php

```diff
@@ -44,8 +44,10 @@ private function applyRoundedStyle(string $blockContent, array $parsedBlock): st
    */
   private function addImageSizeWhenMissing(array $parsedBlock, string $imageUrl): array {
     if (!isset($parsedBlock['attrs']['width'])) {
+      $maxWidth = $parsedBlock['email_attrs']['width'] ?? '660px';
       $imageSize = wp_getimagesize($imageUrl);
-      $parsedBlock['attrs']['width'] = $imageSize ? "{$imageSize[0]}px" : '100%';
+      $imageSize = $imageSize ? "{$imageSize[0]}px" : $maxWidth;
+      $parsedBlock['attrs']['width'] = ($imageSize > $maxWidth) ? $maxWidth : $imageSize;
```

Comment:
Maybe php min function could do the work here as well. I guess it's a little easier to read. Just as suggestion.